### PR TITLE
fix(ingest): Do not crash when chunks are missing in attachments

### PR DESCRIPTION
**Pull request by untitaker**
Generally speaking we want to stop the consumers when processing a message fails, so we can fix the issue and continue processing from the same offset (for as long as we don't have a dead letter queue at least).

In this case however there's nothing we can fix post-mortem because we are somewhat sure we just lost the data.

Fix SENTRY-FK9

### CODE REVIEW

**Review by jan-auer — CHANGES_REQUESTED** [review]
Please also skip over `MissingChunks` in `EventManager.save`:

https://github.com/getsentry/sentry/blob/ea791cb482d6f77481beed41e64ccc52ce10bc65/src/sentry/event_manager.py#L1312

and

https://github.com/getsentry/sentry/blob/ea791cb482d6f77481beed41e64ccc52ce10bc65/src/sentry/event_manager.py#L607

**Comment by jan-auer** [comment]
For minidumps in `sentry.tasks.process_event` I think it's fine if we raise, since we want the event to fail processing.

File: src/sentry/attachments/base.py [hunk 1]
```diff
@@ -142,7 +146,10 @@ def get_data(self, attachment):
         data = []
 
         for key in attachment.chunk_keys:
-            data.append(zlib.decompress(self.inner.get(key, raw=True)))
+            raw_data = self.inner.get(key, raw=True)
+            if raw_data is None:
+                raise MissingAttachmentChunks()
```
Comment (jan-auer) [inline src/sentry/attachments/base.py]: _nit_: Add the `key` into the message for "debugging"? It contains the project id, which could be useful in the future.

File: src/sentry/event_manager.py [hunk 2]
```diff
@@ -601,10 +601,17 @@ def save(self, project_id, raw=False, assume_normalized=False, start_time=None,
...
         # posting to eventstream to make sure all counters and eventstream are
         # incremented for sure.
-        attachments = get_attachments(cache_key, job["event"])
-        for attachment in attachments:
-            key = "bytes.stored.%s" % (attachment.type,)
-            job["event_metrics"][key] = (job["event_metrics"].get(key) or 0) + len(attachment.data)
+        attachments = []
+        for attachment in get_attachments(cache_key, job["event"]):
+            try:
+                key = "bytes.stored.%s" % (attachment.type,)
```
Comment (jan-auer) [inline src/sentry/event_manager.py]: _nit_: Only do `attachment_data = attachment.data` in here like in the consumer, and the rest in `else:`?

**Review by jan-auer — APPROVED** [review]

File: src/sentry/attachments/base.py (hunk 1 above)
Comment (untitaker) [reply src/sentry/attachments/base.py]: This causes very poor grouping. I put the cache key into the wrapping log stmts instead, where we are able to intercept the formatting template

File: src/sentry/event_manager.py (hunk 2 above)
Comment (untitaker) [reply src/sentry/event_manager.py]: done

**Review by tonyo — APPROVED** [review]

File: src/sentry/attachments/base.py (hunk 1 above)
Comment (jan-auer) [reply src/sentry/attachments/base.py]: You could do it by quoting it `"`, in which case we mask it for grouping.

Comment (untitaker) [reply src/sentry/attachments/base.py]: really? where is this happening

Comment (jan-auer) [reply src/sentry/attachments/base.py]: Nvm, you're right. We only do it for hashes, but this won't be enough.

https://github.com/getsentry/sentry/blob/15ea025f431ee7981fdfeca6f7fc47322a32ccb9/src/sentry/grouping/strategies/message.py#L11

Comment (untitaker) [reply src/sentry/attachments/base.py]: It's a really cool idea though. I think we should consider building this

**Pull request merged**